docs/Creating-Components.md

```diff
@@ -0,0 +1,124 @@
+
+### Add a test
+In your component subdirectory, add a `index.test.jsx` to take a "snapshot" so that we can protect against regressions to the component. If the structure of the component changes, the unit test suite will fail.
+
+```js
+import React from 'react';
+import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
+import Blockquote from './index';
+
+describe('Blockquote', () => {
+  shouldMatchSnapshot(
+    'should render correctly',
+    <Blockquote>To be or not to be, that is the question</Blockquote>,
+  );
+});
+```
+
+### Update the schema
+data/schema.yaml describes the Article API definition for web and apps. We need to make it aware of our new component.
+
+Your component is more than likely a new 'block', so you'll need to add it to the array of what blocks are allowed to look like:
+
+```yaml
+    blocks:
+      type: object
+      items:
+        oneOf:
+          - $ref: '#/components/schemas/altText'
+          ... etc ...
+          - $ref: '#/components/schemas/[your component name]'
+      minItems: 2
+```
+
+You'll also need to define the block subtype itself:
+
+```yaml
+    blockquote:
+      type: object
+      required:
+        - model
+        - type
+      properties:
+        model:
+          properties:
+            blocks:
+              $ref: '#/components/schemas/blocks'
+          type: object
+        type:
+          enum:
+            - blockquote
+          type: string
+```
+
+The schema check currently only happens on local data. Eventually, we hope to run the schema validator at runtime.
+
+### Update some local fixture data
+Pick a JSON file under `data/test/news/articles/[id].json`, and:
+
+1) add an example of your block somewhere in the `content.model.blocks` array.
+2) add your new component to the `blockTypes` array.
+
+Run `npm run dev` and you should see your component at your article of choice, eg http://localhost:7080/news/articles/c0000000027o
+
+### Add a container
+We've added our _component_, which should be kept as simple as possible. Now we need to create our _container_, which contains the business logic for mapping Optimo block data to the React parameters our component needs.
+
+Add a new folder under `src/app/containers/[Component Name]/`. You will need:
+
+* index.jsx - describes the mapping of Optimo block data to React parameters
+* index.stories.jsx - describes the behaviour of your component under different data conditions
+
+This step is quite complicated, so copy and paste from a similar example and tweak the code to your requirements.
+
+As part of this work, you'll need to...
+
+### Define your component propTypes
+Under `src/app/models/propTypes/[Component Name (lowercase)]`, create an `index.jsx` which describes the propTypes your component needs. Import this into your `src/app/containers/[Component Name]/index.jsx`.
+
+### Update the Article component
```

Comment:
It would be great to clarify that integrating into the Article component should be done separately to the creation of the component and container. This is because we should conduct a full accessibility review / accessibility swarm as part of the integration ticket.

Reply:
Ok! I didn't know it should 👍 will work on it now

Reply:
Sorry, I should have explained more clearly. 
For PRs 1 & 2, we do some accessibility testing when the PR gets to the UX/Accessibility review column. 
We use this as a reference to guide the accessibility review step https://bbc-news.github.io/accessibility-news-and-you/accessibility-and-supported-assistive-technology

For PR 3, which is integrating into the Article, when the PR gets to the UX/Accessibility review column, we do a full accessibility swarm, with people from multiple disciplines - UX designers, devs, testers, BAs, etc. 

For this, we have a spreadsheet template that is currently in Google Drive. I will get a blank version to reference here in our documentation.